Sources/Environment.swift

```diff
@@ -8,39 +8,22 @@
 import Foundation
 import OrderedCollections
 
-class Environment {
-    var parent: Environment?
+final class Environment: @unchecked Sendable {
+    let parent: Environment?
 
-    var variables: [String: any RuntimeValue] = [
-        "namespace": FunctionValue(value: { args, _ in
-            if args.isEmpty {
-                return ObjectValue(value: [:])
-            }
-            guard args.count == 1, let objectArg = args[0] as? ObjectValue else {
-                throw JinjaError.runtime("`namespace` expects either zero arguments or a single object argument")
-            }
-            return objectArg
-        }),
+    private let lock = NSLock()
+    private var _variables: [String: any RuntimeValue] = [:]
 
-        // Add strftime_now function to handle date formatting in templates
-        "strftime_now": FunctionValue(value: { args, _ in
-            let now = Date()
-
-            if args.count > 0, let formatArg = args[0] as? StringValue {
-                let format = formatArg.value
-
-                let result = formatDate(now, withFormat: format)
-                return StringValue(value: result)
-            }
-
-            // Default format if no arguments
-            let formatter = DateFormatter()
-            formatter.dateFormat = "MMMM dd, yyyy"
-            return StringValue(value: formatter.string(from: now))
-        }),
-    ]
+    var variables: [String: any RuntimeValue] {
+        get {
+            lock.withLock { _variables }
+        }
+        set {
+            lock.withLock { _variables = newValue }
+        }
+    }
```

Comment:
I think we should use NSLock to guard the mutable state.This kept everything synchronous but let's use also make it thread safe.